From 003bfe04a1e90c004506c12cba9dc61ef415726b Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Wed, 23 Sep 2020 19:05:23 +0200
Subject: [PATCH] plc/plc.h: fix build with gcc 10

Add typedef to plcproperty structure to avoid the following build
failure with gcc 10 (which defaults to -fno-common):

/home/buildroot/autobuild/run/instance-3/output-1/host/lib/gcc/arm-buildroot-linux-gnueabihf/10.2.0/../../../../arm-buildroot-linux-gnueabihf/bin/ld: rules.o:(.bss+0x0): multiple definition of `plcproperty'; pibruin.o:(.bss+0x0): first defined here

Fixes:
 - http://autobuild.buildroot.org/results/6b3064b64dea3d4aaf219f787429c25918101483

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
---
 plc/plc.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plc/plc.h b/plc/plc.h
index 1fa32b53..09a72f3f 100644
--- a/plc/plc.h
+++ b/plc/plc.h
@@ -484,7 +484,7 @@ signed WriteParameters2 (struct plc *, unsigned module, const struct nvm_header2
 #pragma pack (push,1)
 #endif
 
-struct __packed plcproperty
+typedef struct __packed plcproperty
 
 {
 	uint8_t PROP_OPTION;
